core/src/main/scala/com.snowplowanalytics.stream/loader/Config.scala

```diff
@@ -38,74 +38,61 @@ import pureconfig.module.enumeratum._
-    case object Bad       extends StreamType
-    case object PlainJson extends StreamType
-
-    val values = findValues
-  }
-
-  sealed trait BadSink extends EnumEntry with EnumEntry.Hyphencase
-  object BadSink extends Enum[BadSink] {
-    case object Stderr  extends BadSink
-    case object Nsq     extends BadSink
-    case object None    extends BadSink
-    case object Kinesis extends BadSink
-
-    val values = findValues
-  }
-
-  sealed trait GoodSink extends EnumEntry with EnumEntry.Hyphencase
-  object GoodSink extends Enum[GoodSink] {
-    case object Elasticsearch extends GoodSink
-    case object Stdout        extends GoodSink
-
-    val values = findValues
-
-    implicit val goodSinkReader: ConfigReader[GoodSink] = deriveEnumerationReader[GoodSink]
-  }
-
-  case class SinkConfig(good: GoodSink, bad: BadSink)
-
-  object SinkConfig {
-    implicit val sinkConfigReader: ConfigReader[SinkConfig] = deriveReader[SinkConfig]
+  implicit def hint[T]: ProductHint[T]               = ProductHint[T](ConfigFieldMapping(CamelCase, CamelCase))
+  implicit val queueHint: FieldCoproductHint[Source] = new FieldCoproductHint[Source]("type")
+
+  case class Monitoring(snowplow: Monitoring.SnowplowMonitoring)
+  object Monitoring {
+    implicit val monitoringConfigReader: ConfigReader[Monitoring] =
+      deriveReader[Monitoring]
+
+    case class SnowplowMonitoring(
+      collector: String,
+      appId: String
+    )
+
+    object SnowplowMonitoring {
+      implicit val snowplowMonitoringConfig: ConfigReader[SnowplowMonitoring] =
+        deriveReader[SnowplowMonitoring]
+    }
   }
 
-  case class AWSConfig(accessKey: String, secretKey: String)
-
-  object AWSConfig {
-    implicit val awsConfigReader: ConfigReader[AWSConfig] = deriveReader[AWSConfig]
-  }
+  sealed trait Source extends Product with Serializable
+  object Source {
+    case class Stdin() extends Source
+    object Stdin {
+      implicit val stdinReader: ConfigReader[Stdin] = deriveReader[Stdin]
+    }
 
-  sealed trait Queue
-  object Queue {
     final case class Nsq(
+      streamName: String,
       channelName: String,
-      nsqdHost: String,
-      nsqdPort: Int,
       nsqlookupdHost: String,
-      nsqlookupdPort: Int
-    ) extends Queue
+      nsqlookupdPort: Int,
+      buffer: Nsq.Buffer
+    ) extends Source
 
     object Nsq {
       implicit val nsqReader: ConfigReader[Nsq] = deriveReader[Nsq]
+
+      case class Buffer(recordLimit: Long)
+      object Buffer {
+        implicit val nsqBufferReader: ConfigReader[Buffer] = deriveReader[Buffer]
+      }
     }
 
     final case class Kinesis(
+      streamName: String,
       initialPosition: String,
       initialTimestamp: Option[String],
       maxRecords: Long,
       region: String,
       appName: String,
       customEndpoint: Option[String],
       dynamodbCustomEndpoint: Option[String],
-      disableCloudWatch: Option[Boolean]
-    ) extends Queue {
+      disableCloudWatch: Option[Boolean],
```

Comment:
In postgres loader, the option to disable cloud watch goes in the `monitoring.metrics` section of the hocon: https://github.com/snowplow-incubator/snowplow-postgres-loader/blob/master/config/config.kinesis.reference.hocon#L86